clients/unshaded/src/main/java/tachyon/client/file/BaseFileSystem.java

```diff
@@ -0,0 +1,276 @@
+    free(path, FreeOptions.defaults());
+  }
+
+  @Override
+  public void free(TachyonURI path, FreeOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      masterClient.free(path, options);
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public URIStatus getStatus(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    return getStatus(path, GetStatusOptions.defaults());
+  }
+
+  @Override
+  public URIStatus getStatus(TachyonURI path, GetStatusOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      return new URIStatus(masterClient.getFileInfo(path));
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public List<URIStatus> listStatus(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    return listStatus(path, ListStatusOptions.defaults());
+  }
+
+  @Override
+  public List<URIStatus> listStatus(TachyonURI path, ListStatusOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    List<FileInfo> fileInfos;
+    try {
+      fileInfos = masterClient.getFileInfoList(path);
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+    List<URIStatus> uriStatuses = new ArrayList<URIStatus>(fileInfos.size());
+    for (FileInfo info : fileInfos) {
+      uriStatuses.add(new URIStatus(info));
+    }
+    return uriStatuses;
+  }
+
+  @Override
+  public TachyonURI loadMetadata(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    return loadMetadata(path, LoadMetadataOptions.defaults());
+  }
+
+  @Override
+  public TachyonURI loadMetadata(TachyonURI path, LoadMetadataOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      return masterClient.loadMetadata(path, options);
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public void mount(TachyonURI src, TachyonURI dst) throws IOException, TachyonException {
+    mount(src, dst, MountOptions.defaults());
+  }
+
+  @Override
+  public void mount(TachyonURI src, TachyonURI dst, MountOptions options)
+      throws IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      masterClient.mount(src, dst);
```

Comment:
currently, the thrift interface does not support options and the client-side MountOptions struct has no options